pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -233,8 +235,8 @@ public void run() {
                 }
             }
         } catch (Throwable t) {
-            log.error("Uncaught exception in Java Instance", t);
-            deathException = (Exception) t;
+            log.error("[{}] Uncaught exception in Java Instance", functionName, t);
+            deathException = (t instanceof Exception) ? (Exception) t : t.getCause();
```

Comment:
I think we should just remove the casting here, since `deathException` is already defined as `Throwable`.

Reply:
oops.. I didn't realize it. fixed it.